plugins/PostProcessingPlugin/scripts/DisplayProgressOnLCD.py

```diff
@@ -26,9 +26,19 @@ def getSettingDataString(self):
                 "time_remaining":
                 {
                     "label": "Time Remaining",
-                    "description": "When enabled, write Time Left: HHMMSS on the display using M117. This is updated every layer.",
-                    "type": "bool",
-                    "default_value": false
+                    "description": "Select to write remaining time on the display using M117 status line message (almost all printers) or using M73 command (Prusa and Marlin 2 if enabled).",
+                    "type": "enum",
```

Comment:
I see that this change from `bool` to `enum` needs a version upgrade, actually. It's a bit much to ask you to develop a version upgrade plug-in so I'll see what I can do.

Reply:
Isn't it way easier to just add a new setting? 

That way there is also no need of a version upgrade script, which makes it a lot cleaner. 

So my proposal is to basicly let the orginal setting remain what it is ("Should time be written at all") and add a new setting that is enabled when the original setting is true ("So now that you want to write time, how should I do it?")

Reply:
I can't really say which way is better. Is there any other reason to bump the version?
Does each plugin have their own version tracking and upgrade system, or is it something global affecting all components?

If the version change is for future 4.9, it would give plugin developers an opportunity to go through their UIs and improve usability, if there is a need.
If this plugin is changed to old checkbox + new controls (I'm slightly worried about usability and wording), would it be on path to 4.8 or 4.9?

Reply:
In either case it would be 4.9

I'm not against bumping the version number, mind you. But I don't really see a very compelling reason to do so in this case. I will discuss this with the other developers as well so find out what they think.

Reply:
It would indeed always be 4.9 since 4.8 is feature-frozen at the moment, with the beta imminent.
I agree that splitting it up into two settings is nicer. I'll just create a checkbox and a drop-down.

Reply:
So it's OK to add settings but NOT OK to change setting types.
Are there restrictions to removing settings? If not, then one could drop existing setting and introduce new setting to go around type change ;)

Reply:
What the version upgrade does is to ensure that people with existing profiles keep their behaviour the same. Dropping the existing setting and adding a new one under a different name effectively resets the setting to its (new) default. So it would still need a version upgrade, or users would see that their setting got magically disabled in the upgrade.